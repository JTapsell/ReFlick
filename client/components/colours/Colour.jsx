import React from 'react'
import { connect } from 'react-redux'

import { Dropdown } from 'react-bootstrap'
import CustomToggle from '../toggle-menu/CustomToggle'
import CustomMenu from '../toggle-menu/CustomMenu'

import { setBrushColour, updateFrameMap } from '../../actions/'

class Colour extends React.Component {
  state = {
    frameMap: '0'
  }

  returnDropdowns (colour, frames) {
    const { dispatch, activeFrame } = this.props
    return frames.map(frame => {
      return <Dropdown.Item
        onClick={() => {
          console.log('Colour:', colour + '\n' + 'Frame:', frame.id)
          this.setState({ frameMap: frame.id.toString() })
          dispatch(updateFrameMap(activeFrame, colour.id, frame.id))
        }}
        key={frame.id + colour}>{frame.name}</Dropdown.Item>
    })
  }

  createColour (colour) {
    const { dispatch } = this.props
    return (
      <div className='colour' style={{ width: 40, height: 40, backgroundColor: `rgb(${colour.r},${colour.g},${colour.b})` }}
        onClick={() => { dispatch(setBrushColour(colour)) }}>
        <div className='framePick' style={{ width: 15, height: 15, backgroundColor: 'white' }}>{this.state.frameMap}</div>
      </div>
    )
  }

  render () {
    const { colour, frames } = this.props

    return (
      <React.Fragment>
        <Dropdown>
          {this.createColour(colour)}
          <Dropdown.Toggle as={CustomToggle} id="dropdown-custom-components">
            <span className="frameSpan"></span>
          </Dropdown.Toggle>
          <Dropdown.Menu >
            {this.returnDropdowns(colour, frames)}
          </Dropdown.Menu>
        </Dropdown>
        <br/>
      </React.Fragment>
    )
  }
}

function mapStateToProps (state) {
  return ({
    activeFrame: state.activeFrame
  })
}

export default connect(mapStateToProps)(Colour)

// TODO: Need to be able to set an active frame ->
// 1. [X] Refactor component into a class to be able to store via state
// 2. [ ] Check whether a frame is set as active
// 2a. [ ] Default value should be no frame (null)
