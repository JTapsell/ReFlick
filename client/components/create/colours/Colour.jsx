import React from 'react'
import { connect } from 'react-redux'

import { Dropdown } from 'react-bootstrap'
import CustomToggle from '../../layout/toggle-menu/CustomToggle'

import { setBrushColour, updateFrameMap } from '../../../actions/'

class Colour extends React.Component {

  returnDropdowns (colour, frames) {
    const { dispatch, activeFrame } = this.props
    return frames.map(frame => {
      return <Dropdown.Item
        onClick={() => {
          if (frame.id === 0) { // Special case for null
            dispatch(updateFrameMap(activeFrame, colour.id, null))
          } else {
            dispatch(updateFrameMap(activeFrame, colour.id, frame.id))
          }
        }}
        key={frame.id + colour}>{frame.name}
      </Dropdown.Item>
    })
  }

  createColour (colour) {
    const { dispatch, target } = this.props
    return (
      <div className='colour' style={{ width: 40, height: 40, backgroundColor: `rgb(${colour.r},${colour.g},${colour.b})` }}
        onClick={() => { dispatch(setBrushColour(colour)) }}>
        <div className='framePick' style={{ width: 15, height: 15, backgroundColor: 'white' }}>{target}</div>
      </div>
    )
  }

  render () {
    const { colour, frames } = this.props
    return (
      <React.Fragment>
        <Dropdown>
          {this.createColour(colour)}
          <Dropdown.Toggle as={CustomToggle} id="dropdown-custom-components">
            <span className="frameSpan"></span>
          </Dropdown.Toggle>
          <Dropdown.Menu >
            {this.returnDropdowns(colour, frames)}
            {/* {this.checkForZero()} */}
          </Dropdown.Menu>
        </Dropdown>
        <br/>
      </React.Fragment>
    )
  }
}

function mapStateToProps (state) {
  return ({
    activeFrame: state.activeFrame
  })
}

export default connect(mapStateToProps)(Colour)
