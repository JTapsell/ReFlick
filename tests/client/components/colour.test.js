import React from 'react'
import { shallow } from 'enzyme'
import Colour from '../../../client/components/colours/Colour'

Colour.prototype.componentDidMount = () => {}
test('test runner is working', () => {
  expect(true).toBeTruthy()
})

// describe('Colour', () => {
//   it('Colour should render correctly in "debug" mode', () => {
//     const component = shallow(<Colour debug />)
//     expect(component).toMatchSnapshot()
//   })
// })
