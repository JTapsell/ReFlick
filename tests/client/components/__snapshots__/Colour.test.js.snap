// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colour Colour should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
